src/plugins/data/public/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -76,16 +78,18 @@ export class IndexPatterns {
     return this.savedObjectsCache.map(obj => obj?.attributes?.title);
   };
 
-  getFields = async (fields: string[], refresh: boolean = false) => {
+  getFields = async (fields: IndexPatternCachedFieldType[], refresh: boolean = false) => {
     if (!this.savedObjectsCache || refresh) {
```

Comment:
Let's try to  use `this.getCache` instead of 

```
  if (!this.savedObjectsCache || refresh) {
      await this.refreshSavedObjectsCache();
    }
    if (!this.savedObjectsCache) {
      return [];
    }
```

Reply:
Thanks, agree, but want to keep enchantments out of scope of this fix pr. 
Maybe in a follow up refactoring.

Reply:
https://github.com/elastic/kibana/issues/56900